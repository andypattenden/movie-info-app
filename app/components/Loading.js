var React = require('react');
var PropTypes = require('prop-types');

require('../styles/Loading.css');

class Loading extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			text: props.text
		}
	}

	render() {
		return (
			<p className='loading'>
				{this.state.text}
			</p>
		)
	}
}

Loading.propTypes = {
	text: PropTypes.string.isRequired
}

Loading.defaultProps = {
	text: 'Loading'
};

module.exports = Loading;
