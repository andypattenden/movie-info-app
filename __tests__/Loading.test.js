import React from 'react';
import Loading from '../app/components/Loading';
import renderer from 'react-test-renderer';

it('renders correctly with default text', () => {
	const tree = renderer.create(
		<Loading />
	).toJSON();

	expect(tree).toMatchSnapshot();
});

it('renders correctly with specified text', () => {
	const tree = renderer.create(
		<Loading text='Downloading...' />
	).toJSON();

	expect(tree).toMatchSnapshot();
});
