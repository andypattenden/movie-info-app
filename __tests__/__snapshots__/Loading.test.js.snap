// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default text 1`] = `
<p
  className="loading"
>
  Loading
</p>
`;

exports[`renders correctly with specified text 1`] = `
<p
  className="loading"
>
  Downloading...
</p>
`;
